src/map/lua/lua_baseentity.cpp

```diff
@@ -180,25 +180,37 @@ inline int32 CLuaBaseEntity::showText(lua_State *L)
 
     CLuaBaseEntity* PLuaBaseEntity = Lunar<CLuaBaseEntity>::check(L, 1);
 
+    bool rotateNPC = true;
+
+    if (!lua_isnil(L, 9) && lua_isboolean(L, 9))
+        rotateNPC = lua_toboolean(L, 9);
+
     if (PLuaBaseEntity != nullptr)
     {
         CBaseEntity* PBaseEntity = PLuaBaseEntity->GetBaseEntity();
         if (PBaseEntity->objtype == TYPE_NPC)
         {
             PBaseEntity->m_TargID = m_PBaseEntity->targid;
-            PBaseEntity->loc.p.rotation = getangle(PBaseEntity->loc.p, m_PBaseEntity->loc.p);
+            if (rotateNPC)
+            {
+                PBaseEntity->loc.p.rotation = getangle(PBaseEntity->loc.p, m_PBaseEntity->loc.p);
```

Comment:
we have another function to do this already. It doesn't need to be in messageSpecial. use `lookAt()`

Reply:
that was allready it there i just added a check to bypass it as some instances you dont want the npc rotating, as the boxes do when a message is displayed

Reply:
/facepalms sorry mybad. sometimes this project gives me a migraine.

Reply:
i feel the pain lol

Reply:
actually this makes them turn their head but not their body. I guess boxes are "just a head". ShowText() annoyingly turns the body in its code.